android/src/com/mapswithme/maps/editor/TimetableFragment.java

```diff
@@ -89,51 +89,40 @@ public boolean onBackPressed()
+      if (!OpeningHours.nativeIsTimetableStringValid(filledTimetables))
+      {
+        new AlertDialog.Builder(getActivity())
+            .setMessage(R.string.editor_correct_mistake)
+            .setPositiveButton(android.R.string.ok, null)
+            .show();
+        return;
+      }
+      mIsAdvancedMode = true;
+      mSwitchMode.setText(R.string.editor_time_advanced);
+      mAdvancedModeFragment = (AdvancedTimetableFragment) attachFragment(mAdvancedModeFragment, AdvancedTimetableFragment.class.getName());
+      mAdvancedModeFragment.setTimetables(filledTimetables);
+    }
     else
-      advancedMode();
-  }
-
-  private void simpleMode()
-  {
-    mIsAdvancedMode = false;
-    mSwitchMode.setText(R.string.editor_time_advanced);
-    final Timetable[] filledTimetables = getFilledTimetables(mAdvancedModeFragment, mAdvancedModeFragment);
-    mSimpleModeFragment = (SimpleTimetableFragment) attachFragment(mSimpleModeFragment, SimpleTimetableFragment.class.getName());
-    mSimpleModeFragment.setTimetables(filledTimetables);
-  }
-
-  private void advancedMode()
-  {
-    mIsAdvancedMode = true;
-    mSwitchMode.setText(R.string.editor_time_simple);
-    final Timetable[] filledTimetables = getFilledTimetables(mSimpleModeFragment, mSimpleModeFragment);
-    mAdvancedModeFragment = (AdvancedTimetableFragment) attachFragment(mAdvancedModeFragment, AdvancedTimetableFragment.class.getName());
-    mAdvancedModeFragment.setTimetables(filledTimetables);
-  }
-
-  private boolean hasFilledTimetables(Fragment fragment)
-  {
-    return fragment != null && fragment.isAdded();
-  }
-
-  @Nullable
-  private Timetable[] getFilledTimetables(Fragment fragment, TimetableProvider provider)
-  {
-    if (!hasFilledTimetables(fragment))
-      return null;
-
-    final Timetable[] timetables = provider.getTimetables();
-    if (timetables == null)
     {
-      new AlertDialog.Builder(getActivity())
-          .setMessage(R.string.editor_correct_mistake)
-          .create();
-      return null;
+      final String filledTimetables = mAdvancedModeFragment != null ? mAdvancedModeFragment.getTimetables()
+                                                                    : OpeningHours.nativeTimetablesToString(OpeningHours.nativeGetDefaultTimetables());
+      if (!OpeningHours.nativeIsTimetableStringValid(filledTimetables))
+      {
+        new AlertDialog.Builder(getActivity())
+            .setMessage(R.string.editor_correct_mistake)
+            .setPositiveButton(android.R.string.ok, null)
+            .show();
+        return;
+      }
+      mIsAdvancedMode = false;
+      mSwitchMode.setText(R.string.editor_time_simple);
+      mSimpleModeFragment = (SimpleTimetableFragment) attachFragment(mSimpleModeFragment, SimpleTimetableFragment.class.getName());
+      mSimpleModeFragment.setTimetables(filledTimetables);
```

Comment:
Да вот не выходит здесь красиво вынести без нормального рефакторинга. Тудушка на вынос фрагментов стоит выше, в объявлении полей, поменяется всё вместе в будущем.
> On Mar 29, 2016, at 5:16 PM, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/editor/TimetableFragment.java <https://github.com/mapsme/omim/pull/2614#discussion_r57730296>:
> 
> > -          .create();
> > -      return null;
> > +      final String filledTimetables = mAdvancedModeFragment != null ? mAdvancedModeFragment.getTimetables()
> > +                                                                    : OpeningHours.nativeTimetablesToString(OpeningHours.nativeGetDefaultTimetables());
> > +      if (!OpeningHours.nativeIsTimetableStringValid(filledTimetables))
> > +      {
> > +        new AlertDialog.Builder(getActivity())
> > +            .setMessage(R.string.editor_correct_mistake)
> > +            .setPositiveButton(android.R.string.ok, null)
> > +            .show();
> > +        return;
> > +      }
> > +      mIsAdvancedMode = false;
> > +      mSwitchMode.setText(R.string.editor_time_simple);
> > +      mSimpleModeFragment = (SimpleTimetableFragment) attachFragment(mSimpleModeFragment, SimpleTimetableFragment.class.getName());
> > +      mSimpleModeFragment.setTimetables(filledTimetables);
> Практически копипаст кода сверху. Стоит вынести в метод.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2614/files/b9b7a07ccb512adcfc1a190ee9bc7bde65172a54..4a195bbeebe113d84b7c9fa4c3c254dcbfbbbe3c#r57730296>